sdk/storage/azblob/internal/exported/user_delegation_credential.go

```diff
@@ -0,0 +1,59 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package exported
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/internal/generated"
+)
+
+// NewUserDelegationCredential creates a new UserDelegationCredential using a Storage account's Name and a user delegation Key from it
+func NewUserDelegationCredential(accountName string, udk UserDelegationKey) *UserDelegationCredential {
+	return &UserDelegationCredential{
+		accountName:       accountName,
+		userDelegationKey: udk,
+	}
+}
+
+// UserDelegationKey contains UserDelegationKey.
+type UserDelegationKey = generated.UserDelegationKey
+
+// UserDelegationCredential contains an account's name and its user delegation key.
+type UserDelegationCredential struct {
+	accountName       string
+	userDelegationKey UserDelegationKey
+}
+
+// AccountName returns the Storage account's Name
+func (f *UserDelegationCredential) AccountName() string {
```

Comment:
Will customers need to call this, or can it be removed?  It can always be added later if needed.

Reply:
I do use it in my example test, but I could create a helper method for it (similar to GetUDKParams) or simply access the AccountName from the UserDelegationCredential itself.

Reply:
It's actually used in sas/service.go (line 193) and sas/account.go (112) to get canonical name:

`stringToSign := strings.Join([]string{
		userDelegationCredential.AccountName(),`

Reply:
If this is only needed internally, I think you can use an internal-only helper to get the value (just like for `getUDKParams`).